### ISSUE TITLE
Artificial colored edges in CairoMakie when saving an image to PDF.

### ISSUE THREAD

**Issue by fatteneder** [MakieOrg/Makie.jl#2162]
### MWE

```julia
using CairoMakie
CairoMakie.activate!()
using FileIO
using Makie

img = load(Makie.assetpath("doge.png"))
i = image(permutedims(img)[:,end:-1:1])
save(joinpath(@__DIR__, "mwe.pdf"), i)
```

Output: 
![image](https://user-images.githubusercontent.com/26469701/180289323-3a7d98ae-59f1-463f-9fcf-f4528bdb76ea.png)

Adding `fxaa=true` to `image` does not fix it either.
Saving as `png` does work though.

Related: #2080

**Comment by fatteneder** [comment]
Could this be an issue with a transparent background in the `png` file?

**Comment by jkrumbiegel** [comment]
Have you looked at this with multiple different pdf viewers? Just to make sure we're not looking at an implementation-dependent artifact here. Or is the artifact really baked into a bitmap?

**Comment by fatteneder** [comment]
I checked with
- Evince
- Okular
- Firefox
- Chromium
and all show the same artifacts.

Here is the PDF file if you want to check yourself: [mwe.pdf](https://github.com/JuliaPlots/Makie.jl/files/9162295/mwe.pdf)

**Comment by fatteneder** [comment]
Fixed by #2304
